### PR TITLE
memory desc compatability conditions update

### PR DESCRIPTION
**Pull request by chenhu-wang**
### Details:
 - *update memory description compatibility conditions*

### Tickets:
 - *68125*
 - *66103*

### CODE REVIEW

**Comment by andrei-kochin** [comment]
@maxnick could you please comment?

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_pooling_node.cpp [hunk 1]
```diff
@@ -273,11 +273,8 @@ void MKLDNNPoolingNode::initSupportedPrimitiveDescriptors() {
                 dataConfig.inPlace = -1;
                 dataConfig.constant = false;
                 auto desc = getSrcMemDesc(itpd, i);
-                if (desc->getType() & MemoryDescType::Blocked) {
-                    dataConfig.desc = desc->as<BlockedMemoryDesc>()->cloneWithUndefStridesAndOffset();
-                } else {
-                    dataConfig.desc = std::move(desc);
-                }
+                dataConfig.desc = std::move(desc);
```
Comment (mandrono) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_pooling_node.cpp:276]: why did you remove undefined descs creation for pooling node? I guess we need to restore it.

**Comment by maxnick** [comment]
Follow up task for tests has been created see ticket 71088

**Comment by chenhu-wang** [comment]
closed as included into [PR](https://github.com/openvinotoolkit/openvino/pull/8361): https://github.com/openvinotoolkit/openvino/pull/8361
